checker/src/org/checkerframework/checker/regex/RegexQualifiedTransfer.java

```diff
@@ -45,82 +59,218 @@ public RegexQualifiedTransfer(QualAnalysis<QualParams<Regex>> analysis) {
+        // Look for: constant < mat.groupCount()
+        // Make mat be @Regex(constant + 1)
+        TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> res;
+        res = super.visitLessThan(n, in);
+        res = handleMatcherGroupCount(n.getRightOperand(), n.getLeftOperand(), false, in, res);
+        return res;
+    }
+
+    @Override
+    public
+    TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>>
+    visitLessThanOrEqual(LessThanOrEqualNode n,
+            TransferInput<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> in) {
+        // Look for: constant <= mat.groupCount()
+        // Make mat be @Regex(constant)
+        TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> res;
+        res = super.visitLessThanOrEqual(n, in);
+        res = handleMatcherGroupCount(n.getRightOperand(), n.getLeftOperand(), true, in, res);
+        return res;
+    }
+
+    @Override
+    public
+    TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>>
+    visitGreaterThan(GreaterThanNode n,
+            TransferInput<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> in) {
+        // Look for: mat.groupCount() > constant
+        // Make mat be @Regex(constant + 1)
+        TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> res;
+        res = super.visitGreaterThan(n, in);
+        res = handleMatcherGroupCount(n.getLeftOperand(), n.getRightOperand(), false, in, res);
+        return res;
+    }
+
+    @Override
+    public
+    TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>>
+    visitGreaterThanOrEqual(GreaterThanOrEqualNode n,
+            TransferInput<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> in) {
+        // Look for: mat.groupCount() >= constant
+        // Make mat be @Regex(constant)
+        TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>> res;
+        res = super.visitGreaterThanOrEqual(n, in); 
+        res = handleMatcherGroupCount(n.getLeftOperand(), n.getRightOperand(), true, in, res);
+        return res;
+    }
+
+    /**
+     * See whether possibleMatcher is a call of groupCount on a Matcher and
+     * possibleConstant is a constant.
+     * If so, annotate the matcher as
+     * constant + 1 if !isAlsoEqual
+     * constant if isAlsoEqual
+     *
+     * @param possibleMatcher the Node that might be a call of Matcher.groupCount()
+     * @param possibleConstant the Node that might be a constant
+     * @param isAlsoEqual whether the comparison operation is strict or reflexive
+     * @param in the input TransferResult
+     * @return the possibly refined output TransferResult
+     */
+    private
+    TransferResult<QualValue<QualParams<Regex>>, QualStore<QualParams<Regex>>>
+    handleMatcherGroupCount(
+            Node possibleMatcher, Node possibleConstant, boolean isAlsoEqual,
```

Comment:
I think "isAlsoEqual" is true if there was a <= test rather than a < test in the source code being checked, that led to this method being called by the checker.  Is that right?